src/install_pypi/plan/planner.rs

```diff
@@ -235,10 +200,86 @@ impl InstallPlanner {
+            Theirs,
+        }
+
+        // Walk over all installed packages and check if they are required
+        let mut extraneous = HashMap::new();
+        for dist in site_packages.iter() {
+            let pkg = required_pkgs.get(dist.name());
+            let installer = dist
+                .installer()
+                .map_or(String::new(), |f| f.unwrap_or_default());
+
+            match pkg {
+                // Apparently we need this package
+                // and we have installed
+                Some(_) => {}
+                // Ignore packages not managed by pixi
+                None if installer != consts::PIXI_UV_INSTALLER => {
+                    // Do check for doubles though
+                    extraneous
+                        .entry(dist.name())
+                        .or_insert(Vec::new())
+                        .push(Extraneous::Theirs);
+                }
+                // Uninstall unneeded packages
+                None => {
+                    match extraneous.entry(dist.name()) {
+                        // We have already seen this package
+                        Entry::Occupied(mut occupied_entry) => {
+                            occupied_entry.get_mut().push(Extraneous::Ours(dist));
+                        }
+                        // This is a new package
+                        Entry::Vacant(vacant_entry) => {
+                            vacant_entry.insert(vec![Extraneous::Ours(dist)]);
+                        }
+                    };
+                }
+            }
+        }
+        // So it may happen that both conda and PyPI have installed a package with the same name
+        // but different versions, in that case, we want to split into extraneous and duplicates
+        let (extraneous, duplicates): (Vec<_>, Vec<_>) =
+            extraneous.into_iter().partition_map(|(_, dists)| {
+                if dists.len() > 1 {
+                    Either::Right(
+                        dists
+                            .into_iter()
+                            .filter_map(|d| {
+                                if let Extraneous::Ours(dist) = d {
+                                    Some(dist.clone())
+                                } else {
+                                    None
+                                }
+                            })
+                            .collect::<Vec<_>>(),
+                    )
+                } else {
+                    Either::Left(
+                        dists
+                            .into_iter()
+                            .filter_map(|d| {
+                                if let Extraneous::Ours(dist) = d {
+                                    Some(dist.clone())
+                                } else {
+                                    None
+                                }
+                            })
+                            .collect::<Vec<_>>(),
+                    )
+                }
+            });
```

Comment:
This is the new removal method which introduces the concept of duplicates.

Reply:
What this logic does, is basically count the number of owned and unowned pacakges. If we find an owned package an no extra un-owned we put it in extraneous, others in duplicates. You can see what happens with this below.

Reply:
Thanks for the explainer!